File: JS/algo.js

 
 
/**
  * Function :
  * @param n integer
  */

function factoriel(n){
           
        var i = 0;
        var fact=1;
        while(i<n){
            i=i+1;
            fact=fact*i;
            console.log(n+"!="+fact);        
}

 }

factoriel(4);

/**
 *  Algoritme de tri
 */

function permuter(tab, i,j){
        
    var a =tab[i];  // la position i je la recupere dans une variable a
    tab[i]=tab[j];  // la position j je la met dans la position i
    tab[j]=a;       // la position i je la met dans la position j au travers de la variable intermediare a 
    console.log(tab);        
 }

 tab = [1,2,3,4,5,8];
 permuter(tab,1,2);

File: src/Algo.java
import java.io.IOException;

public class Algo {

	
	/**
	 * Factoriel 
	 * @param integer n
	 */
	public static  void  factoriel(int n) 
	{
		int incfac=0;
		int fac=1;
		while(incfac<n)
		{
			incfac=incfac+1;
			fac=fac*incfac;	
			System.out.println("***************************************************************************"); 
	        System.out.println(n+"!="+fac); 
		}	
	}
	
	/**
	 * permuter 
	 * @param array t, int i , int j
	 */
	public static void permuter(int[]t, int i, int j) {
           
		 int a = t[i];   
		 t[i]=t[j];     
     	 t[j]= a;      
     	 
     	 for(int k=0;i<t.length;k++) 
     	 {	
 		          System.out.println(t[k]); 
     	 }
	}
	public static void main (String[] args ) throws IOException 
	{
		int[] t={4,3,6,1,2,8};
	
		 //permuter(t, 0, 3);
    
		// Trie a bulles 
        
        for(int k=1;k<t.length;k++) 
   	      {	
        	if(t[k-1]>t[k])
        	{
              // permuter(t, k-1, k);      
        	}      	
   	     }
        factoriel(4);
	}
}
